etcdserver/api/v3rpc/lease.go

```diff
@@ -41,5 +44,24 @@ func (ls *LeaseServer) LeaseRevoke(ctx context.Context, rr *pb.LeaseRevokeReques
 }
 
 func (ls *LeaseServer) LeaseKeepAlive(stream pb.Lease_LeaseKeepAliveServer) error {
-	panic("not implemented")
+	for {
+		req, err := stream.Recv()
+		if err == io.EOF {
+			return nil
+		}
+		if err != nil {
+			return err
+		}
+
+		ttl, err := ls.le.LeaseRenew(lease.LeaseID(req.ID))
+		if err != nil {
+			return ErrLeaseNotFound
```

Comment:
ErrNotPrimary is a temp error. We should remove it after we implement forwarding logic. But I agree that we should never mask it. Probably we can panic it for now.